subsys/net/ip/ipv6_mld.c

```diff
@@ -183,6 +183,14 @@ int net_ipv6_mld_join(struct net_if *iface, const struct in6_addr *addr)
 	struct net_if_mcast_addr *maddr;
 	int ret;
 
+	if (!net_if_is_up(iface)) {
+		return -ENETDOWN;
+	}
+
+	if (net_if_flag_is_set(iface, NET_IF_IPV6_NO_MLD)) {
```

Comment:
IMO that check should remain where it was. Ok to skip address registration if the interface is down, but if MLD protocol is disabled on an interface, I think we should still register the address, w/o sending a report. Some modules like mDNS use  MLD API to register mcast addresses, so skipping mcast address registration here would make them unusable with OpenThread.

Reply:
Is there a use-case where one wants to add an mcast address, but not proceed with MLD even if its iface enables MLD? (i.e. an mcast address that would not fall into the MLD scope, if that does exist?)

@jukkar  do you know?

Just asking, because if not: it would be less counter-intuitive to call MLD in net_if_ipv6_maddr_add(), and above examples would call net_if_ipv6_maddr_add() instead.

Reply:
> it would be less counter-intuitive to call MLD in net_if_ipv6_maddr_add(), and above examples would call net_if_ipv6_maddr_add() instead.

I 100% agree on that.

Reply:
Dunno, the mcast functions were there first and I suppose we could add multicast addresses if needed without joining the group. But to join the multicast group, the MLD/IGMP API adds the multicast address and then joins the group.